// @dotkomonline/design-system/dist/components/message/Message.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const styled_components_1 = __importStar(require("styled-components"));
const colors_1 = require("../../common/colors");
const Icon_1 = __importDefault(require("../icon/Icon"));
const IconWrapper = styled_components_1.default.span `
  color: ${({ color }) => color};
  font-size: 1.5rem;
  margin-right: 0.5rem;
  display: flex;
`;
const StyledBox = styled_components_1.default.div `
  display: flex;
  padding: 1rem;
  border-radius: 3px;
  line-height: 1.25;
  ${({ status }) => styled_components_1.css `
    border: 2px solid ${colors_1.colors[status]};
    color: ${colors_1.colors[`${status}Darken70`]};
    background: ${colors_1.colors[`${status}Lighten90`]};
  `}
`;
const Message = ({ status, children }) => {
    const iconNames = {
        info: 'info',
        success: 'done',
        warning: 'warning',
        error: 'clear',
    };
    return (React.createElement(StyledBox, { status: status },
        React.createElement(IconWrapper, { color: colors_1.colors[status] },
            React.createElement(Icon_1.default, { name: iconNames[status] })),
        children));
};
exports.default = Message;
