// @dotkomonline/design-system/dist/components/typography/Heading.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const styled_components_1 = __importStar(require("styled-components"));
const HeadingStyle = styled_components_1.css `
  margin: 4rem 0 2rem;
  line-height: 1.5;

  &:first-child {
    margin-top: 0;
  }

  &:last-child {
    margin-bottom: 0;
  }
`;
const Heading = styled_components_1.default.h1 `
  font-size: var(--font-size-xxl);
  line-height: 6rem;
  ${HeadingStyle}
`;
exports.Heading = Heading;
const Heading2 = styled_components_1.default.h2 `
  font-size: calc(var(--font-size-xl));
  line-height: 3.75rem;
  ${HeadingStyle}
`;
exports.Heading2 = Heading2;
const Heading3 = styled_components_1.default.h3 `
  font-size: calc(var(--font-size-lg));
  line-height: 3.125rem;
  ${HeadingStyle}
`;
exports.Heading3 = Heading3;
const Heading4 = styled_components_1.default.h4 `
  font-size: calc(var(--font-size-md));
  line-height: 2.5rem;
  font-weight: 900;
  ${HeadingStyle}
`;
exports.Heading4 = Heading4;
const Heading5 = styled_components_1.default.h5 `
  font-size: calc(var(--font-size-sm));
  line-height: 2rem;
  font-weight: 900;
  ${HeadingStyle}
`;
exports.Heading5 = Heading5;
const Heading6 = styled_components_1.default.h6 `
  font-size: calc(var(--font-size-xs));
  line-height: 2rem;
  font-weight: 900;
  ${HeadingStyle}
`;
exports.Heading6 = Heading6;
