// @dotkomonline/design-system/dist/components/typography/Markdown.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const markdown_to_jsx_1 = __importDefault(require("markdown-to-jsx"));
const Heading_1 = require("./Heading");
const Text_1 = require("./Text");
exports.Markdown = ({ children, options = {} }) => {
    return (react_1.default.createElement(markdown_to_jsx_1.default, { options: {
            overrides: {
                h1: { component: Heading_1.Heading },
                h2: { component: Heading_1.Heading2 },
                h3: { component: Heading_1.Heading3 },
                h4: { component: Heading_1.Heading4 },
                h5: { component: Heading_1.Heading5 },
                h6: { component: Heading_1.Heading6 },
                p: { component: Text_1.Paragraph },
                blockquote: { component: Text_1.Quote },
            },
            ...options,
        } }, children));
};
exports.md = (literals, ...placeholders) => {
    const result = [];
    for (let i = 0; i < placeholders.length; i++) {
        result.push(literals[i]);
        result.push(placeholders[i]);
    }
    result.push(literals[literals.length - 1]);
    return react_1.default.createElement(exports.Markdown, null, result.join(''));
};
