// @dotkomonline/design-system/dist/common/colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const blender_1 = require("./blender");
const primary = '#0D5474';
const secondary = '#F9B759';
const white = '#FFFFFF';
const gray = '#808080';
const black = '#000000';
const info = '#0A94C2';
const success = '#09AA09';
const warning = '#FFC000';
const error = '#DF2020';
exports.colors = {
    primaryLighten45: blender_1.lighten(primary, 0.45),
    primaryLighten30: blender_1.lighten(primary, 0.3),
    primaryLighten15: blender_1.lighten(primary, 0.15),
    primary: primary,
    primaryDarken15: blender_1.darken(primary, 0.15),
    primaryDarken30: blender_1.darken(primary, 0.3),
    primaryDarken45: blender_1.darken(primary, 0.45),
    secondaryLighten45: blender_1.lighten(secondary, 0.45),
    secondaryLighten30: blender_1.lighten(secondary, 0.3),
    secondaryLighten15: blender_1.lighten(secondary, 0.15),
    secondary: secondary,
    secondaryDarken15: blender_1.darken(secondary, 0.15),
    secondaryDarken30: blender_1.darken(secondary, 0.3),
    secondaryDarken45: blender_1.darken(secondary, 0.45),
    white: white,
    grayLighten90: blender_1.lighten(gray, 0.9),
    grayLighten60: blender_1.lighten(gray, 0.6),
    grayLighten30: blender_1.lighten(gray, 0.3),
    gray: gray,
    grayDarken30: blender_1.darken(gray, 0.3),
    grayDarken60: blender_1.darken(gray, 0.6),
    grayDarken90: blender_1.darken(gray, 0.9),
    black: black,
    infoLighten90: blender_1.lighten(info, 0.9),
    infoLighten45: blender_1.lighten(info, 0.45),
    info: info,
    infoDarken35: blender_1.darken(info, 0.35),
    infoDarken70: blender_1.darken(info, 0.7),
    successLighten90: blender_1.lighten(success, 0.9),
    successLighten45: blender_1.lighten(success, 0.45),
    success: success,
    successDarken35: blender_1.darken(success, 0.35),
    successDarken70: blender_1.darken(success, 0.7),
    warningLighten90: blender_1.lighten(warning, 0.9),
    warningLighten45: blender_1.lighten(warning, 0.45),
    warning: warning,
    warningDarken35: blender_1.darken(warning, 0.35),
    warningDarken70: blender_1.darken(warning, 0.7),
    errorLighten90: blender_1.lighten(error, 0.9),
    errorLighten45: blender_1.lighten(error, 0.45),
    error: error,
    errorDarken35: blender_1.darken(error, 0.35),
    errorDarken70: blender_1.darken(error, 0.7),
    eventsBedpres: '#eb536e',
    eventsKurs: '#127dbd',
    eventsSosialt: '#43b171',
    eventsUtflukt: '#fdbd47',
    eventsEkskursjon: '#2ac6f9',
    eventsKjelleren: '#e75e3b',
    eventsInternt: '#e75e3b',
    eventsAnnet: '#b36bcd',
    officialBlue: '#0D5474',
    officialOrange: '#F9B759',
};
