// @dotkomonline/design-system/dist/index.js
"use strict";
/**
 * Add all public components as exports here to allow
 * importing them as `import { Component } from "package-name";`
 */
Object.defineProperty(exports, "__esModule", { value: true });
const colors_1 = require("./common/colors");
exports.colors = colors_1.colors;
const global_1 = require("./common/global");
exports.GlobalStyle = global_1.GlobalStyle;
const layout_1 = require("./common/layout");
exports.media = layout_1.media;
exports.breakPoints = layout_1.breakPoints;
// components/message
var Message_1 = require("./components/message/Message");
exports.Message = Message_1.default;
// components/button
var Button_1 = require("./components/button/Button");
exports.Button = Button_1.default;
// components/card
var Card_1 = require("./components/card/Card");
exports.Card = Card_1.default;
// components/checkbox
var Checkbox_1 = require("./components/checkbox/Checkbox");
exports.Checkbox = Checkbox_1.default;
// components/eventCard
var EventCard_1 = require("./components/eventCard/EventCard");
exports.EventCard = EventCard_1.default;
// components/icon
var Icon_1 = require("./components/icon/Icon");
exports.Icon = Icon_1.default;
// component/link
var Link_1 = require("./components/link/Link");
exports.Link = Link_1.default;
// component/logo
var Logo_1 = require("./components/logo/Logo");
exports.Logo = Logo_1.default;
var CompactLogo_1 = require("./components/logo/CompactLogo");
exports.CompactLogo = CompactLogo_1.default;
var OldLogo_1 = require("./components/logo/OldLogo");
exports.OldLogo = OldLogo_1.default;
var OldCompactLogo_1 = require("./components/logo/OldCompactLogo");
exports.OldCompactLogo = OldCompactLogo_1.default;
var ModernLogo_1 = require("./components/logo/ModernLogo");
exports.ModernLogo = ModernLogo_1.default;
var StaticSpinnerLogo_1 = require("./components/logo/StaticSpinnerLogo");
exports.StaticSpinnerLogo = StaticSpinnerLogo_1.default;
// component/modal
var Modal_1 = require("./components/modal/Modal");
exports.Modal = Modal_1.default;
// component/radio
var RadioGroup_1 = require("./components/radio/RadioGroup");
exports.RadioGroup = RadioGroup_1.default;
var RadioButton_1 = require("./components/radio/RadioButton");
exports.RadioButton = RadioButton_1.default;
// component/select
var Select_1 = require("./components/select/Select");
exports.Select = Select_1.default;
// component/spinner
var Spinner_1 = require("./components/spinner/Spinner");
exports.Spinner = Spinner_1.default;
var SpinnerCircles_1 = require("./components/spinner/SpinnerCircles");
exports.SpinnerCircles = SpinnerCircles_1.default;
var SpinnerLogo_1 = require("./components/spinner/SpinnerLogo");
exports.SpinnerLogo = SpinnerLogo_1.default;
// components/tab
var Tab_1 = require("./components/tabs/Tab");
exports.Tab = Tab_1.default;
var Tabs_1 = require("./components/tabs/Tabs");
exports.Tabs = Tabs_1.default;
// component/textArea
var TextArea_1 = require("./components/textArea/TextArea");
exports.TextArea = TextArea_1.default;
// component/textField
var TextField_1 = require("./components/textField/TextField");
exports.TextField = TextField_1.default;
// components/timeline
var Timeline_1 = require("./components/timeline/Timeline");
exports.Timeline = Timeline_1.default;
// component/toggleSwitch
var ToggleSwitch_1 = require("./components/toggleSwitch/ToggleSwitch");
exports.ToggleSwitch = ToggleSwitch_1.default;
// component/typography
var Heading_1 = require("./components/typography/Heading");
exports.Heading = Heading_1.Heading;
exports.Heading2 = Heading_1.Heading2;
exports.Heading3 = Heading_1.Heading3;
exports.Heading4 = Heading_1.Heading4;
exports.Heading5 = Heading_1.Heading5;
exports.Heading6 = Heading_1.Heading6;
var Text_1 = require("./components/typography/Text");
exports.Ingress = Text_1.Ingress;
exports.Paragraph = Text_1.Paragraph;
exports.Quote = Text_1.Quote;
exports.Info = Text_1.Info;
var Markdown_1 = require("./components/typography/Markdown");
exports.Markdown = Markdown_1.Markdown;
exports.md = Markdown_1.md;
