// @dotkomonline/design-system/dist/components/button/Button.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importStar(require("styled-components"));
const colors_1 = require("../../common/colors");
const StyledButton = styled_components_1.default.button `
  appearance: none;
  background: ${(props) => props.color};
  border: 2px solid ${(props) => props.color};
  color: ${colors_1.colors.white};
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
  border-radius: 3px;
  padding: 0.5rem 1rem;
  cursor: pointer;

  &:hover {
    background-image: linear-gradient(0, #0001 0, #0001 0);
  }

  &:disabled {
    background: ${colors_1.colors.grayLighten90};
    color: ${colors_1.colors.grayLighten60};
    box-shadow: none;
    border-color: ${colors_1.colors.grayLighten90};
    cursor: not-allowed;
  }

  ${(props) => props.variant == 'outline' &&
    styled_components_1.css `
      background: transparent;
      border-color: ${props.color};
      color: ${props.color};
    `}
`;
const Button = ({ value = '', children = null, color = 'primary', variant = 'normal', disabled = false, ...props }) => {
    let mainColor = colors_1.colors.primary;
    switch (color) {
        case 'secondary': {
            mainColor = colors_1.colors.secondary;
            break;
        }
        case 'success': {
            mainColor = colors_1.colors.success;
            break;
        }
        case 'danger': {
            mainColor = colors_1.colors.error;
            break;
        }
    }
    const extraProps = {
        color: mainColor,
        disabled,
        variant,
    };
    return (react_1.default.createElement(StyledButton, Object.assign({}, props, extraProps), children || value));
};
exports.default = Button;
