// @dotkomonline/design-system/dist/components/eventCard/EventCard.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const colors_1 = require("../../common/colors");
const Icon_1 = __importDefault(require("../icon/Icon"));
const Card_1 = __importDefault(require("../card/Card"));
const EventInfo = styled_components_1.default.div `
  display: flex;
  align-items: center;
  justify-content: flex-end;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.1);
  min-height: 4rem;
  width: 100%;
  border-radius: 3px;
  background: #fff;
`;
const ColorCodeIndicator = styled_components_1.default.span `
  display: inline-block;
  background: ${({ color }) => color || colors_1.colors.primary};
  width: 7px;
  min-height: calc(4rem - 20px);
  margin: 10px 0;
  border-radius: 2px;
`;
const Text = styled_components_1.default.p `
  display: inline;
  margin-left: 5px;
`;
const Box = styled_components_1.default.span `
  display: inline-flex;
  align-items: center;
  margin: 0 15px;

  &:first-child {
    margin-right: auto;
  }
`;
const Picture = styled_components_1.default.img `
  width: 100%;
  border-bottom: 0.5px solid #b7b7b7;
  &:hidden {
    display: none;
  }
`;
const CategoryInfo = styled_components_1.default.h2 `
  background: ${({ color }) => color || colors_1.colors.primary};
  color: #fff;
  margin: 0;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  padding: 7px 0;
  text-align: center;
  font-size: 20px;
  font-weight: 200;
  letter-spacing: 1px;
  &:hidden {
    display: none;
  }
`;
const EventCard = ({ eventName, eventColor, date, attending, imgSrc = '', isLargeEvent, category }) => (react_1.default.createElement(Card_1.default, { padding: false },
    react_1.default.createElement(CategoryInfo, { color: eventColor, hidden: !isLargeEvent }, category),
    react_1.default.createElement(Picture, { src: imgSrc, hidden: !isLargeEvent }),
    react_1.default.createElement(EventInfo, null,
        react_1.default.createElement(Box, null,
            react_1.default.createElement(ColorCodeIndicator, { color: eventColor }),
            react_1.default.createElement(Text, null, eventName)),
        react_1.default.createElement(Box, null,
            react_1.default.createElement(Icon_1.default, { name: "date_range" }),
            react_1.default.createElement(Text, null, date)),
        react_1.default.createElement(Box, null,
            react_1.default.createElement(Icon_1.default, { name: "perm_identity" }),
            react_1.default.createElement(Text, null, attending)))));
exports.default = EventCard;
