// @dotkomonline/design-system/dist/components/colors/Colors.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const colors_1 = require("../../common/colors");
const StyledColor = styled_components_1.default.div `
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  background: ${({ color }) => colors_1.colors[color]};
  border: 2px solid ${({ color }) => colors_1.colors[color]};
  color: ${({ darkText }) => (darkText ? colors_1.colors.black : colors_1.colors.white)};
  width: 100%;
  padding: 1rem;
  transition: background 0.2s ease-out, border-color 0.1s;

  &:hover {
    border-color: ${({ darkText }) => (darkText ? colors_1.colors.black : colors_1.colors.white)};
  }
`;
const ColorSwatch = ({ color, darkText = false }) => {
    return (React.createElement(StyledColor, { color: color, darkText: darkText },
        React.createElement("span", null, color),
        React.createElement("samp", null, colors_1.colors[color])));
};
exports.default = ColorSwatch;
