// @dotkomonline/design-system/dist/components/link/Link.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importStar(require("styled-components"));
const colors_1 = require("../../common/colors");
const StyledLink = styled_components_1.default.a `
  position: relative;
  text-decoration: none;
  color: ${colors_1.colors.primary};
  transition: color 0.1s;
  cursor: pointer;

  &::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 1px;
    background: ${colors_1.colors.primary};
    bottom: 0;
    left: 0;
    transition: transform 0.1s, color 0.1s;
    ${(props) => !props.underline &&
    styled_components_1.css `
        transform: scaleX(0);
      `}
  }

  &:hover,
  &:focus {
    outline: none;
    color: ${colors_1.colors.primaryLighten30};

    &::after {
      transform: scaleX(1);
      background: ${colors_1.colors.primaryLighten30};
    }
  }
`;
const Link = ({ href, underline = false, children }) => {
    return (react_1.default.createElement(StyledLink, { href: href, underline: underline }, children));
};
exports.default = Link;
