// @dotkomonline/design-system/dist/components/textField/TextField.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const colors_1 = require("../../common/colors");
const Icon_1 = __importDefault(require("../icon/Icon"));
const statuses = {
    error: colors_1.colors.error,
    warning: colors_1.colors.warning,
    success: colors_1.colors.success,
};
const InputField = styled_components_1.default.input `
  width: 100%;
  border-radius: 3px;
  padding: 0.5rem;
  padding-right: 2.5rem;
  font-size: 1em;
  color: ${colors_1.colors.black};
  border: 2px solid ${({ status }) => (status ? statuses[status] : colors_1.colors.grayLighten60)};
  transition: border-color 0.2s;
  box-sizing: border-box;

  &:focus {
    outline: none;
    border-color: ${colors_1.colors.primary};
  }

  &:disabled {
    background-color: ${colors_1.colors.grayLighten90};
    color: ${colors_1.colors.grayDarken30};
  }

  &:invalid {
    border-color: ${({ status }) => (status ? statuses[status] : colors_1.colors.error)};
  }
`;
const InputLabel = styled_components_1.default.label `
  display: block;
  font-size: 0.8rem;
  margin-bottom: 0.5rem;
  color: ${colors_1.colors.primary};

  &:empty {
    display: none;
  }
`;
const InputMessage = styled_components_1.default.p `
  font-size: 0.8rem;
  margin-top: 0.5rem;
  color: ${colors_1.colors.error};

  &:empty {
    display: none;
  }
`;
const ClearButton = styled_components_1.default.button `
  position: absolute;
  padding: 0.8rem;
  font-size: 1rem;
  top: 0;
  right: 0;
  bottom: 0;
  color: ${colors_1.colors.grayLighten30};
  cursor: pointer;
  background: transparent;
`;
const RelativeContainer = styled_components_1.default.div `
  position: relative;
`;
const StyledIcon = styled_components_1.default(Icon_1.default) `
  display: block;
`;
const ClearableInputField = ({ disabled, ...props }) => {
    const [text, setText] = react_1.useState(props.defaultValue || '');
    return (react_1.default.createElement(RelativeContainer, null,
        react_1.default.createElement(InputField, Object.assign({ value: text, onChange: (e) => {
                setText(e.target.value);
            } }, props, { disabled: disabled })),
        text && !disabled && (react_1.default.createElement(ClearButton, { onClick: () => setText('') },
            react_1.default.createElement(StyledIcon, { name: "clear" })))));
};
const TextField = ({ type = 'text', label = '', errorMessage = '', ...props }) => {
    return (react_1.default.createElement("div", null,
        react_1.default.createElement(InputLabel, null, label),
        react_1.default.createElement(ClearableInputField, Object.assign({ type: type }, props)),
        react_1.default.createElement(InputMessage, null, errorMessage)));
};
exports.default = TextField;
