// @dotkomonline/design-system/dist/components/timeline/Day.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const colors_1 = require("../../common/colors");
const layout_1 = require("../../common/layout");
const Day = ({ children, day }) => {
    return (react_1.default.createElement(StyledDay, null,
        react_1.default.createElement(DayString, null, day),
        react_1.default.createElement(StyledDayBody, null, children)));
};
exports.default = Day;
const StyledDay = styled_components_1.default.div `
  &:not(:first-child) {
    margin-top: 1rem;
  }

  @media ${layout_1.media.desktopAndUp} {
    display: flex;
  }
`;
const DayString = styled_components_1.default.h2 `
  color: ${colors_1.colors.primaryDarken15};
  font-size: 1.25rem;
  margin: 0;
  padding: 1rem 2rem 1rem 0;
  width: 275px;
  white-space: nowrap;
  &::first-letter {
    text-transform: uppercase;
  }
`;
const StyledDayBody = styled_components_1.default.div `
  position: relative;

  &::before {
    content: '';
    display: block;
    background: ${colors_1.colors.primary};
    width: 2px;
    height: calc(100% - 3.1rem);
    top: 1.5rem;
    position: absolute;
    left: calc(5rem - 1px);
  }
`;
