// @dotkomonline/design-system/dist/components/timeline/Event.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const colors_1 = require("../../common/colors");
const Event = ({ title, startTime, endTime, children }) => {
    const [expanded, setExpanded] = react_1.useState(false);
    const toggleExpanded = () => {
        setExpanded(!expanded);
    };
    return (react_1.default.createElement(StyledEvent, null,
        react_1.default.createElement(EventHeader, { onClick: toggleExpanded },
            react_1.default.createElement(EventTime, null, startTime),
            react_1.default.createElement(EventTitle, null, title)),
        react_1.default.createElement(EventBody, { hide: !expanded },
            react_1.default.createElement(EventInnerContent, null, children),
            react_1.default.createElement(EventEndTime, null,
                "Antatt sluttidspunkt: ",
                endTime))));
};
exports.default = Event;
const widthTime = '5rem';
const StyledEvent = styled_components_1.default.div ``;
const EventTitle = styled_components_1.default.h3 `
  position: relative;
  padding: 1rem 0 1rem 2rem;

  &::before {
    content: '';
    display: inline-block;
    background-color: ${colors_1.colors.primary};
    border-radius: 50%;
    width: 1rem;
    height: 1rem;
    position: absolute;
    left: 0;
    top: 50%;
    transform: translate(-50%, -50%);

    ${StyledEvent}:hover & {
      background-color: ${colors_1.colors.secondary};
    }
  }
`;
const EventHeader = styled_components_1.default.div `
  display: flex;
  position: relative;
  cursor: pointer;
`;
const EventTime = styled_components_1.default.h3 `
  padding: 1rem 0;
  width: ${widthTime};
`;
const EventEndTime = styled_components_1.default.p `
  text-decoration: underline;
  margin-top: 1rem;
`;
const EventBody = styled_components_1.default.div `
  padding: 0.5rem 2rem 1rem;
  margin-left: ${widthTime};
  max-width: 670px;
  line-height: 1.25;

  ${({ hide }) => hide &&
    `
    display: none;
  `};
`;
const EventInnerContent = styled_components_1.default.div `
  & > * + * {
    margin-top: 1rem;
  }
`;
