// @dotkomonline/design-system/dist/components/spinner/SpinnerCircles.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importStar(require("styled-components"));
const colors_1 = require("../../common/colors");
const SpinnerContainer = styled_components_1.default.div `
  position: relative;
  width: ${({ size }) => size};
  height: ${({ size }) => size};
`;
const cssloadPartAnim = styled_components_1.keyframes `
  0% {
    transform: translateX(-50%);
  }
  100% {
    transform: translateX(50%);
  }
`;
const swapZIndex = styled_components_1.keyframes `
  from {
    z-index: auto;
  }
  to {
    z-index: 10;
  }
`;
const SpinnerCircle = styled_components_1.css `
  position: absolute;
  border-radius: 50%;
  z-index: 1;
  animation: ${cssloadPartAnim} 0.92s cubic-bezier(0.445, 0.05, 0.55, 0.95) infinite alternate;
`;
const SpinnerPrimaryCircle = styled_components_1.default.div `
  ${SpinnerCircle};
  left: 50%;
  width: ${({ size }) => size};
  height: ${({ size }) => size};
  background: ${({ color }) => color};
  animation: ${cssloadPartAnim} 0.92s cubic-bezier(0.445, 0.05, 0.55, 0.95) infinite alternate-reverse,
    ${swapZIndex} 0.92s steps(2) -0.46s infinite alternate;
`;
const SpinnerSecondaryCircle = styled_components_1.default.div `
  ${SpinnerCircle};
  left: 50%;
  width: ${({ size }) => size};
  height: ${({ size }) => size};
  background: ${({ color }) => color};
`;
const SpinnerCircles = ({ primaryColor = colors_1.colors.primary, secondaryColor = colors_1.colors.primaryLighten30, size = '2rem', }) => (react_1.default.createElement(SpinnerContainer, { size: size },
    react_1.default.createElement(SpinnerPrimaryCircle, { size: size, color: primaryColor }),
    react_1.default.createElement(SpinnerSecondaryCircle, { size: size, color: secondaryColor })));
exports.default = SpinnerCircles;
