// @dotkomonline/design-system/dist/components/spinner/Spinner.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const colors_1 = require("../../common/colors");
const SpinnerSVG = styled_components_1.default.svg `
  width: 50px;
`;
const Path = styled_components_1.default.path.attrs({
    d: 'M25,8.2C15.72,8.2,8.2,15.72,8.2,25',
}) ``;
const SpinnerPath = styled_components_1.default(Path) `
  fill: none;
  stroke-width: 6;
  stroke-linecap: round;
  stroke-miterlimit: 10;
  transform: rotate(${({ start }) => start % 361}deg);
  transform-origin: 50% 50%;
  animation-fill-mode: forwards;

  stroke: ${({ color }) => color};
  animation: ${({ time }) => time}s spin infinite;
  animation-timing-function: cubic-bezier(0.43, 0.54, 0.61, 0.74);

  @keyframes spin {
    from {
      transform: rotate(${({ start }) => start % 361}deg);
    }
    to {
      transform: rotate(${({ end }) => end % 361}deg);
    }
  }
`;
const Spinner = ({ primaryColor = colors_1.colors.primary, secondaryColor = colors_1.colors.secondary }) => {
    return (react_1.default.createElement(SpinnerSVG, { version: "1.1", x: "0px", y: "0px", viewBox: "0 0 50 50" },
        react_1.default.createElement(SpinnerPath, { time: 1.4, start: 5, end: 5, color: secondaryColor }),
        react_1.default.createElement(SpinnerPath, { time: 1.2, start: 0, end: 360, color: primaryColor })));
};
exports.default = Spinner;
