// @dotkomonline/design-system/dist/components/select/Select.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const react_select_1 = __importDefault(require("react-select"));
const styled_components_1 = __importDefault(require("styled-components"));
const colors_1 = require("../../common/colors");
const StyledSelect = styled_components_1.default(react_select_1.default) `
  transition: all 0.2s;
  padding: 8px 20px 8px 8px;
  background: #fff;
  border-radius: 3px;
  border: 2px solid ${colors_1.colors.grayLighten60};

  cursor: pointer;

  &:hover {
    border-color: ${colors_1.colors.primary};
  }
  appearance: none;
`;
const Select = ({ options, ...rest }) => {
    return React.createElement(StyledSelect, Object.assign({ options: options }, rest));
};
exports.default = Select;
